test/health_check_test_case.py

```diff
@@ -34,124 +40,181 @@ def _other_lambda_names(self) -> List[str]:
+            **self._expected_api_endpoints(endpoint_states)
+        }, health_object)
+
+    @mock_sts
+    @mock_sqs
+    def test_one_api_endpoint_down(self):
+        self._create_mock_queues()
+        endpoint_states = self._generate_endpoint_states(self.endpoints[1:], down_endpoints=self.endpoints[:1])
+        response = self._test(endpoint_states, lambdas_up=True)
+        health_object = response.json()
+        self.assertEqual(503, response.status_code)
+        self.assertEqual({
+            'up': False,
+            'elastic_search': {
+                'up': True
+            },
+            **self._expected_queues(True),
+            **self._expected_other_lambdas(True),
+            **self._expected_api_endpoints(endpoint_states)
         }, health_object)
 
     @mock_sts
     @mock_sqs
     def test_elasticsearch_down(self):
         self._create_mock_queues()
         mock_endpoint = ('nonexisting-index.com', 80)
+        endpoint_states = self._generate_endpoint_states(self.endpoints)
         with mock.patch.dict(os.environ, **config.es_endpoint_env(mock_endpoint)):
-            response = self._test(up=True)
+            response = self._test(endpoint_states, lambdas_up=True)
             health_object = response.json()
             self.assertEqual(503, response.status_code)
             documents_ = {
                 'up': False,
                 'elastic_search': {
                     'up': False
                 },
-                'queues': {
-                    'up': True,
-                    **({
-                        queue_name: {'up': True, 'messages': {'delayed': 0, 'invisible': 0, 'queued': 0}}
-                        for queue_name in config.all_queue_names
-                    })
-                },
-                **({
-                    lambda_name: {'up': True}
-                    for lambda_name in self._other_lambda_names()
-                }),
-                'unindexed_bundles': 0,
-                'unindexed_documents': 0
+                **self._expected_queues(True),
+                **self._expected_other_lambdas(True),
+                **self._expected_api_endpoints(endpoint_states)
             }
             self.assertEqual(documents_, health_object)
 
-    def _test(self, up: bool):
+    def _expected_queues(self, up: bool) -> JSON:
+        return {
+            'queues': {
+                'up': up,
+                **({
+                    queue_name: {
+                        'up': True,
+                        'messages': {
+                            'delayed': 0, 'invisible': 0, 'queued': 0
+                        }
+                    } if up else {
+                        'up': False, 'error': 'The specified queue does not exist for this wsdl version.'
+                    }
+                    for queue_name in config.all_queue_names
+                })
+            }
+        }
+
+    def _expected_api_endpoints(self, endpoint_states: Dict[str, bool]) -> JSON:
```

Comment:
Don't use Dict, use Mapping (or MutableMapping if the function modifies the argument which I don't think it does).